buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/performance/generator/tasks/RemoteProject.groovy

```diff
@@ -125,12 +134,13 @@ class RemoteProject extends DefaultTask {
         }
     }
 
-    private void checkoutRef(String remoteUri, String ref, File tmpDir) {
-        project.exec {
+    @TypeChecked(TypeCheckingMode.SKIP)
+    private static void checkoutRef(Task task, String remoteUri, String ref, File tmpDir) {
+        task.project.exec {
```

Comment:
Why don't you pass in `project` instead of `task`? I think the static method should have no knowledge about the `task.`

Reply:
Done.